fvm/blueprints/fees.go

```diff
@@ -0,0 +1,165 @@
+  }
+}
+`
+
+func DeployTxFeesContractTransaction(service, fungibleToken, flowToken, flowFees flow.Address) *flow.TransactionBody {
+	contract := contracts.FlowFees(
+		fungibleToken.HexWithPrefix(),
+		flowToken.HexWithPrefix(),
+	)
+
+	return flow.NewTransactionBody().
+		SetScript([]byte(fmt.Sprintf(deployTxFeesTransactionTemplate, hex.EncodeToString(contract)))).
+		AddAuthorizer(flowFees).
+		AddAuthorizer(service)
+}
+
+const deployStorageFeesTransactionTemplate = `
+transaction {
+  prepare(serviceAccount: AuthAccount) {
+    serviceAccount.contracts.add(name: "FlowStorageFees", code: "%s".decodeHex())
+  }
+}
+`
+
+func DeployStorageFeesContractTransaction(service flow.Address, contract []byte) *flow.TransactionBody {
+	return flow.NewTransactionBody().
+		SetScript([]byte(fmt.Sprintf(deployStorageFeesTransactionTemplate, hex.EncodeToString(contract)))).
+		AddAuthorizer(service)
+}
+
+const getFlowTokenAvailableBalanceScriptTemplate = `
+import FlowStorageFees from 0x%s
+
+pub fun main(): UFix64 {
+  return FlowStorageFees.defaultTokenAvailableBalance(0x%s)
+}
+`
+
+func GetFlowTokenAvailableBalanceScript(accountAddress, serviceAddress flow.Address) []byte {
+	return []byte(fmt.Sprintf(getFlowTokenAvailableBalanceScriptTemplate, serviceAddress, accountAddress))
+}
+
+const getStorageCapacityScriptTemplate = `
+import FlowStorageFees from 0x%s
+
+pub fun main(): UFix64 {
+	return FlowStorageFees.calculateAccountCapacity(0x%s)
+}
+`
+
+func GetStorageCapacityScript(accountAddress, serviceAddress flow.Address) []byte {
+	return []byte(fmt.Sprintf(getStorageCapacityScriptTemplate, serviceAddress, accountAddress))
+}
+
+const setupFeesTransactionTemplate = `
+import FlowStorageFees, FlowServiceAccount from 0x%s
+
+transaction(transactionFee: UFix64, accountCreationFee: UFix64, minimumStorageReservation: UFix64, storageMegaBytesPerReservedFLOW: UFix64) {
+    prepare(service: AuthAccount) {
+        let serviceAdmin = service.borrow<&FlowServiceAccount.Administrator>(from: /storage/flowServiceAdmin)
+            ?? panic("Could not borrow reference to the flow service admin!");
+
+        let storageAdmin = service.borrow<&FlowStorageFees.Administrator>(from: /storage/storageFeesAdmin)
+            ?? panic("Could not borrow reference to the flow storage fees admin!");
+
+        serviceAdmin.setTransactionFee(transactionFee)
+        serviceAdmin.setAccountCreationFee(accountCreationFee)
+        storageAdmin.setMinimumStorageReservation(minimumStorageReservation)
+        storageAdmin.setStorageMegaBytesPerReservedFLOW(storageMegaBytesPerReservedFLOW)
+    }
+}
+`
+
+func SetupFeesTransaction(
+	service flow.Address,
+	transactionFee,
+	addressCreationFee,
+	minimumStorageReservation,
+	storagePerFlow cadence.UFix64,
+) *flow.TransactionBody {
+	transactionFeeArg, err := jsoncdc.Encode(transactionFee)
+	if err != nil {
+		panic(fmt.Sprintf("failed to encode transaction fee: %s", err.Error()))
```

Comment:
I get that this path should really trigger panic, but shouldn't we return error here and let the caller decide how to handle it?

Reply:
since it is only called at deployment time I think is fine, it is also a lot of code change as we wrap this blue prints most of the time :(